src/quest_manager/models.py

```diff
@@ -975,3 +989,142 @@ def get_minutes_to_complete(self):
 
         minutes = (self.first_time_completed - self.timestamp).total_seconds() / 60
         return minutes
+
+
+class QuestionType(models.TextChoices):
+    """Identifies what types of questions can be created.
+
+    This is used in the views and forms to determine what type of question is being created, and to show/hide specific
+    fields. For example, with a short answer question, the solution_file field is hidden, but with a file upload question,
+    the solution_file field is shown.
+    """
+
+    SHORT_ANSWER = "short_answer", "Short Answer"
+    LONG_ANSWER = "long_answer", "Long Answer"
+    FILE_UPLOAD = "file_upload", "File Upload"
+
+
+class Question(models.Model):
+    """Used for adding additional submission fields to Quests.
+
+    This model is used for all types of questions, including multiple choice, short answer, file upload, etc.
+    """
+
+    instructions = models.TextField(
+        help_text=(
+            "Appears to the student when they are completing the quest. Should provide "
+            "specific guidance, a prompt, or a problem statement for the student to address."
+        )
+    )
+    solution_text = models.TextField(
+        null=True,
+        blank=True,
+        help_text=(
+            "This text will appear to the marker, allowing for an easy comparison of "
+            "the student's submission to an ideal solution."
+        ),
+    )
+    solution_file = RestrictedFileField(
+        blank=True,
+        null=True,
+        upload_to="quest/question/solution/%Y/%m/%d",
+        help_text=(
+            "This file will appear to the marker, allowing them to easily compare "
+            "the student's submission to the ideal solution for file-based questions."
+        ),
+    )
+
+    # Each tuple contains the file type as both the stored value and the display value
+    # e.g. ("media", "Media") or ("image", "Image")
+    allowed_file_type = models.CharField(
+        max_length=50,
+        choices=[(key, key.capitalize()) for key in FILE_MIME_TYPES.keys()],
+        default="media",
+        help_text=("The types of files that can be uploaded by a student for this question.")
+    )
+
+    marker_notes = models.TextField(
+        null=True,
+        blank=True,
+        help_text=(
+            "This text will appear to the marker, providing additional details about "
+            "what a student's submission should be or how the question should be marked."
+        ),
+    )
+    required = models.BooleanField(
+        default=True,
+        help_text=(
+            "If this is checked, the student must complete this "
+            "question to complete the quest."
+        ),
+    )
+    datetime_created = models.DateTimeField(auto_now_add=True)
+    datetime_last_edit = models.DateTimeField(auto_now=True)
+    ordinal = models.PositiveIntegerField(
+        default=1,
+        help_text=(
+            "Indicates the order of the question in the quest so that "
```

Comment:
"Questions will be displayed to students in order from lowest ordinal to highest ordinal."

Reply:
```py
    datetime_created = models.DateTimeField(auto_now_add=True)
    datetime_last_edit = models.DateTimeField(auto_now=True)
    ordinal = models.PositiveIntegerField(
        default=1,
        help_text=(
            "Questions will be displayed to the student in order from lowest ordinal to "
            "highest ordinal."
        ),
    )
```